<a href="https://colab.research.google.com/github/cheongmoi/nous-agent/blob/master/machine-learning/Iris_dataset.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## 查看鸢尾花数据集的特征信息

我们可以通过以下几种方式查看鸢尾花数据集的特征结构、特征名称和详细描述。

In [ ]:
from sklearn.datasets import load_iris
import pandas as pd

# 加载鸢尾花数据集
iris = load_iris()

# 1. 查看特征名称 (Feature Names)
print("数据集中的特征名称:")
for name in iris.feature_names:
    print(f" - {name}")

# 2. 查看目标分类名称 (Target Names)
print("\n分类目标名称:")
print(iris.target_names)

# 3. 查看数据集的详细描述信息 (Dataset Description)
# 这里我们只打印前 20 行描述信息，你可以打印完整的 iris.DESCR 获取详细说明
print("\n数据集描述信息 (前 20 行):")
print("\n".join(iris.DESCR.split("\n")[:20]))

数据集中的特征名称:
 - sepal length (cm)
 - sepal width (cm)
 - petal length (cm)
 - petal width (cm)

分类目标名称:
['setosa' 'versicolor' 'virginica']

数据集描述信息 (前 20 行):
.. _iris_dataset:

Iris plants dataset
--------------------

**Data Set Characteristics:**

:Number of Instances: 150 (50 in each of three classes)
:Number of Attributes: 4 numeric, predictive attributes and the class
:Attribute Information:
    - sepal length in cm
    - sepal width in cm
    - petal length in cm
    - petal width in cm
    - class:
            - Iris-Setosa
            - Iris-Versicolour
            - Iris-Virginica

:Summary Statistics:


## 鸢尾花 (Iris) 数据集详细介绍

鸢尾花数据集（Iris dataset）是机器学习和统计学中最著名的经典数据集之一，由英国统计学家和生物学家 **罗纳德·费希尔 (Ronald Fisher)** 于 1936 年在其论文中首次发表。

该数据集常用于多类分类（Multi-class Classification）任务的基准测试。

### 1. 数据集组成
数据集共包含 **150 个样本**，分为 **3 个品种**（每个品种各有 50 个样本）。

### 2. 特征信息 (Features)
每个样本有 **4 个数值型特征**（单位均为厘米 $\text{cm}$）：
*   **萼片长度 (Sepal Length)**：花萼的长度。
*   **萼片宽度 (Sepal Width)**：花萼的宽度。
*   **花瓣长度 (Petal Length)**：花瓣的长度。
*   **花瓣宽度 (Petal Width)**：花瓣的宽度。

### 3. 分类目标 (Targets)
数据集的目标是将鸢尾花精准分类到以下三个品种之一：
1.  **Setosa (山鸢尾)**：通常花瓣较小，在数学上非常容易与其他两个品种区分开（线性可分）。
2.  **Versicolor (变色鸢尾)**：特征介于山鸢尾和维吉尼亚鸢尾之间。
3.  **Virginica (维吉尼亚鸢尾)**：通常花瓣较大。

### 4. 为什么这个数据集如此受欢迎？
*   **规模适中**：只有 150 个样本，非常适合用于初学者快速上手、算法原型设计和教学演示。
*   **易于可视化**：由于特征只有 4 个，可以通过绘制散点图（Scatter Plot）或成对关系图（Pairplot）直观地观察不同类别之间的分布关系。
*   **经典的分类挑战**：山鸢尾（Setosa）是线性可分的，而变色鸢尾（Versicolor）和维吉尼亚鸢尾（Virginica）在特征空间中存在重叠，需要更复杂的决策边界来划分，这很好地展示了不同机器学习模型的分类能力。